examples/autoignition-mpi.py

```diff
@@ -52,73 +55,103 @@
+    initialize_logmgr,
+    logmgr_add_many_discretization_quantities,
+    logmgr_add_device_name,
+    logmgr_add_device_memory_usage,
+    set_sim_state
+)
+
 import cantera
 import pyrometheus as pyro
 
 logger = logging.getLogger(__name__)
 
 
 @mpi_entry_point
-def main(ctx_factory=cl.create_some_context, casename="autoignition", use_leap=False,
-         restart_step=None, restart_name=None):
+def main(ctx_factory=cl.create_some_context, use_logmgr=False,
+         use_leap=False, use_profiling=False, casename="autoignition",
+         rst_step=None, rst_name=None):
     """Drive example."""
     cl_ctx = ctx_factory()
-    queue = cl.CommandQueue(cl_ctx)
-    actx = PyOpenCLArrayContext(queue,
+
+    if casename is None:
+        casename = "mirgecom"
+
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    nproc = comm.Get_size()
+
+    logmgr = initialize_logmgr(use_logmgr,
+        filename=f"{casename}.sqlite", mode="wu", mpi_comm=comm)
+
+    if use_profiling:
+        queue = cl.CommandQueue(cl_ctx,
+            properties=cl.command_queue_properties.PROFILING_ENABLE)
+        actx = PyOpenCLProfilingArrayContext(queue,
+            allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)),
+            logmgr=logmgr)
+    else:
+        queue = cl.CommandQueue(cl_ctx)
+        actx = PyOpenCLArrayContext(queue,
             allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)))
 
+    # Some discretization parameters
     dim = 2
     nel_1d = 8
     order = 1
 
+    # {{{ Time stepping control
+
     # This example runs only 3 steps by default (to keep CI ~short)
     # With the mixture defined below, equilibrium is achieved at ~40ms
     # To run to equlibrium, set t_final >= 40ms.
+
+    # Time stepper selection
+    if use_leap:
+        from leap.rk import RK4MethodBuilder
+        timestepper = RK4MethodBuilder("state")
+    else:
+        timestepper = rk4_step
+
+    # Time loop control parameters
+    current_step = 0
     t_final = 1e-8
     current_cfl = 1.0
-    velocity = np.zeros(shape=(dim,))
     current_dt = 1e-9
     current_t = 0
     constant_cfl = False
+
+    # i.o frequencies
     nstatus = 1
     nviz = 5
+    nhealth = 1
     nrestart = 5
-    rank = 0
-    checkpoint_t = current_t
-    current_step = 0
-    if use_leap:
-        from leap.rk import RK4MethodBuilder
-        timestepper = RK4MethodBuilder("state")
-    else:
-        timestepper = rk4_step
-    box_ll = -0.005
-    box_ur = 0.005
-    error_state = False
+
+    # }}}  Time stepping control
+
     debug = False
 
-    from mpi4py import MPI
-    comm = MPI.COMM_WORLD
-    rank = comm.Get_rank()
-    nproc = comm.Get_size()
+    rst_path = "restart_data/"
+    rst_pattern = (
+        rst_path + "{cname}-{step:04d}-{rank:04d}.pkl"
+    )
+    if rst_step:  # read the grid from restart data
+        rst_fname = rst_pattern.format(cname=casename, step=rst_step, rank=rank)
```

Comment:
I made some tweaks to the restart code to reduce the amount of restart/non-restart conditional handling: https://github.com/majosm/mirgecom/commit/9a6ff93144205666fd62b7b40178f07ffaa927f8. Thoughts?

Reply:
The problem here is that this precludes the user from specifying the restart filename. You want the user to be able to specify an arbitrary file for restart.  When it comes to *writing* the restart files, those can definitely be constructed using the casename - but the user needs to be able to specify the restart filename at restart time.

Reply:
Should be fixable now that i know what `rst_name` is for -- but I think we can make this cleanup a separate PR.